server/eclipse-project/src/main/java/us/freeandfair/corla/query/BallotManifestInfoQueries.java

```diff
@@ -154,4 +155,35 @@ public static OptionalLong count() {
     
     return result;
   }
+
+  // public static Set<BallotManifestInfo> where (cond) {}
+
+  /**
+     Find the batch(bmi) that would hold the sequence number given.
+   */
+  public static Optional<BallotManifestInfo> holdingSequenceNumber(final Long the_n) {
+    Set<BallotManifestInfo> result = null;
+
+    try {
+      final Session s = Persistence.currentSession();
+      final CriteriaBuilder cb = s.getCriteriaBuilder();
+      final CriteriaQuery<BallotManifestInfo> cq =
+          cb.createQuery(BallotManifestInfo.class);
+      final Root<BallotManifestInfo> root = cq.from(BallotManifestInfo.class);
+      final List<Predicate> disjuncts = new ArrayList<Predicate>();
+      final Predicate start = cb.greaterThanOrEqualTo(root.get("my_sequence_start"), the_n);
+      final Predicate end = cb.lessThanOrEqualTo(root.get("my_sequence_end"), the_n);
+      disjuncts.add(start);
+      disjuncts.add(end);
+      cq.select(root).where(cb.or(disjuncts.toArray(new Predicate[disjuncts.size()])));
+      final TypedQuery<BallotManifestInfo> query = s.createQuery(cq);
+      result = new HashSet<BallotManifestInfo>(query.getResultList());
+    } catch (final PersistenceException e) {
+      Main.LOGGER.error("Exception when reading ballot manifests from database: " + e);
```

Comment:
Should we start passing the exception as the second argument to these logger functions so their traces etc. have a chance to get logged? I notice it's not done elsewhere in the file but it feels right when logging exceptions.

Reply:
done